test/e2e/specs/specs-playwright/wp-likes-spec.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import config from 'config';
+import {
+	BrowserHelper,
+	DataHelper,
+	LoginFlow,
+	NewPostFlow,
+	GutenbergEditorPage,
+	LikesComponent,
+} from '@automattic/calypso-e2e';
+
+/**
+ * Constants
+ */
+const mochaTimeOut = config.get( 'mochaTimeoutMS' );
+const host = DataHelper.getJetpackHost();
+const viewportName = BrowserHelper.getViewportName();
+const quote =
+	'The foolish man seeks happiness in the distance. The wise grows it under his feet.\n— James Oppenheim';
+
+describe( `[${ host }] Likes: (${ viewportName }) @parallel`, function () {
+	this.timeout( mochaTimeOut );
+
+	describe( 'New post', function () {
+		let gEditor;
+		let likesComponent;
+
+		before( 'Can log in', async function () {
+			const loginFlow = new LoginFlow( this.page, 'gutenbergSimpleSiteUser' );
+			await loginFlow.login();
+		} );
+
+		it( 'Start new post', async function () {
+			const newPostFlow = new NewPostFlow( this.page );
+			await newPostFlow.newPostFromNavbar();
+		} );
+
+		it( 'Enter post title', async function () {
+			gEditor = await GutenbergEditorPage.Expect( this.page );
+			const title = DataHelper.randomPhrase();
+			await gEditor.enterTitle( title );
+			await gEditor.enterText( quote );
+		} );
+
+		it( 'Publish and visit post', async function () {
+			await gEditor.publish( true );
+		} );
+
+		it( 'Like post', async function () {
+			likesComponent = await LikesComponent.Expect( this.page );
+			await likesComponent.clickLike();
```

Comment:
How will we know the post was actually liked?

Reply:
Similar to [this](https://github.com/Automattic/wp-calypso/pull/53169/files#r639619316), would be nice here to have i.e. a `likesComponent.isLiked()` method to verify the state and add a necessary assertion.

Reply:
In the current implementation, I suppose there isn't an explicit check for if the request came back and acknowledged the click on the like button.

I assume the `Promise.race` in `clickLike` function serves as an implicit check on if the like button was indeed clicked, but it may not be enough. I will see if there's a way to make the check more explicit.

Reply:
So, this was bit of a struggle as I had the right idea, but each time had just _something_ wrong with my approach.

I was able to come up with a solution based on if the Like button has already been clicked or not, then checking for an event to be fired to the appropriate endpoint (which depend on if we're Liking or Unliking the post).

The Playwright logs confirm that my solution appears to work properly by detecting the request event.

Correct run
```
  pw:api   checking visibility of "text="You like this."" +2ms
  pw:api <= frame.isVisible succeeded +21ms
  pw:api => frame.click started +0ms
  pw:api => page.waitForEvent started +3ms   <--- wait for the request to a new like endpoint
  ...
  pw:api attempting click action +3ms
  pw:api   waiting for element to be visible, enabled and stable +0ms
  pw:api   element is visible, enabled and stable +16ms
  pw:api   scrolling into view if needed +0ms
  pw:api   done scrolling +1ms
  pw:api   checking that element receives pointer events at (316.22,748) +3ms
  pw:api   element does receive pointer events +3ms
  pw:api   performing click action +0ms
  pw:api   click action done +8ms
  pw:api   waiting for scheduled navigations to finish +0ms
  pw:api <= page.waitForEvent succeeded +1ms   <--- request resolved
  pw:api   navigations have finished +1ms
  pw:api <= frame.click succeeded +0ms
```

Intentionally incorrect run
```
  pw:api waiting for event "request" +1ms   <---  wait for request to endpoint that I intentionally mangled
  pw:api attempting click action +2ms
  pw:api   waiting for element to be visible, enabled and stable +0ms
  pw:api   element is visible, enabled and stable +6ms
  pw:api   scrolling into view if needed +1ms
  pw:api   done scrolling +0ms
  pw:api   checking that element receives pointer events at (316.22,748) +1ms
  pw:api   element does receive pointer events +2ms
  pw:api   performing click action +0ms
  pw:api   click action done +8ms
  pw:api   waiting for scheduled navigations to finish +1ms
  pw:api   navigations have finished +1ms
  pw:api <= frame.click succeeded +1ms
  pw:api   "load" event fired +2s
  pw:api   "domcontentloaded" event fired +1ms
  pw:api   "load" event fired +10ms
  pw:api   "networkidle" event fired +1s
  pw:api <= page.waitForEvent failed +11s  <--- event never resolved (as I would expect) leading to test fail.
```

Reply:
TIL that the request events are supported in Playwright, nice! 👍 It looks good, though we're relying on the request URLs now. I was thinking that we could maybe simplify it and just check if the button's state changed, which is how a user would confirm a successful action? Something like:

```js
async clickLike() {
  await this.frame.click( selectors.likeButton );
  await this.frame.waitForSelectort( selectors.likedButton );
}
```
What do you think?

Reply:
I simplified the logic and verified it still works 👍